apps/condo/domains/billing/schema/resolvers/resolver.js

```diff
@@ -0,0 +1,120 @@
+const { isEmpty, isNil, xorWith, isEqual } = require('lodash')
+
+const { GQLError } = require('@open-condo/keystone/errors')
+const { getLogger } = require('@open-condo/keystone/logging')
+
+const isArrayEqual = (x, y) => isEmpty(xorWith(x, y, isEqual))
+
+const appLogger = getLogger('condo')
+
+class Resolver {
+
+    constructor (billingContext, context, params = {}) {
+        const { name = 'resolver' } = params
+        this.name = name
+        this.dvSender = {
+            dv: 1,
+            sender: { dv: 1, fingerprint: `register-receipt-${name}-resolver` },
+        }
+        this.billingContext = billingContext
+        this.context = context
+        this.timeStart = process.hrtime()
+        this.debugMessages = []
+        this.unTouched = 0
+        this.created = 0
+        this.updated = 0
+        this.logger = appLogger.child({ module: 'register-billing-receipts' })
+    }
+
+    error (error, index, errorDescription) {
+        if (errorDescription) {
+            this.logger.error({ msg: `${this.name} internal error`, payload: { errorDescription } })
+        }
+        return new GQLError({ ...error, inputIndex: index }, this.context)
+    }
+
+    debug (message) {
+        this.debugMessages.push(message)
+    }
+
+    async init () {
+        // some init logic
+    }
+
+    buildCreateInput (data, relations = []) {
+        const createInput = { ...this.dvSender }
+        for (const [propertyName, propertyValue] of Object.entries(data)) {
+            if (!isNil(propertyValue)) {
+                if (relations.indexOf(propertyName) !== -1) {
+                    createInput[propertyName] = { connect: { id: propertyValue } }
+                } else {
+                    createInput[propertyName] = propertyValue
+                }
+            }
+        }
+        this.created++
+        this.logger.info({ msg: `Create new ${this.name}`, payload: createInput })
```

Comment:
Look like too match logs. You can't correlate this message to reqId or anything else. It looks useles